connector/src/test/scala/com/basho/riak/spark/rdd/connector/RiakConnectorTests.scala

```diff
@@ -30,16 +46,153 @@ class RiakConnectorTests {
 
   @Test
   def openSessionWithHostsShouldReturnSessionFromPool(): Unit = {
+    mockSessionCache(2, new RiakClient(mock(classOf[RiakCluster])))
     val hosts = HostAndPort.hostsFromString("2.2.2.2", 8780).toSeq
 
-    val session_1 = connector_1.openSession(Some(hosts))
+    val session_1 = connector.openSession(Some(hosts))
     val client_1 = session_1.unwrap()
 
     session_1.close()
 
-    val session_2 = connector_1.openSession(Some(hosts))
+    val session_2 = connector.openSession(Some(hosts))
     val client_2 = session_2.unwrap()
 
     assertTrue(client_1 eq client_2)
   }
+
+  @Test
+  def openSessionWithEqualHostsShouldReturnSessionFromPool(): Unit = {
+    mockSessionCache(3, new RiakClient(mock(classOf[RiakCluster])))
+
+    val hosts = conf.hosts.toSeq
+    val fSession = connector.openSession()
+    val sSession = connector.openSession(Some(hosts))
+
+    assertNotSame(conf.hosts, hosts)
+    assertEquals(conf.hosts, hosts.toSet)
+    assertSame(fSession.unwrap(), sSession.unwrap())
+  }
+
+  @Test
+  def openSessionWithNotEqualHostsShouldReturnDifferentSessions(): Unit = {
+    mockSessionCache(2, new RiakClient(mock(classOf[RiakCluster])))
+
+    val firstSession = connector.openSession()
+    val secondSession = connector.openSession()
+    val thirdSession = connector.openSession(Some(HostAndPort.hostsFromString("2.2.2.2", 0).toSeq))
+
+    assertSame(firstSession.unwrap(), secondSession.unwrap())
```

Comment:
Not exactly the same. The aim of this test is check session eviction. Because the size of the cache is 2 and only third session opening should cause cause evictio